markets/storageadapter/provider.go

```diff
@@ -45,14 +46,25 @@ type ProviderNodeAdapter struct {
 
 	secb *sectorblocks.SectorBlocks
 	ev   *events.Events
+
+	jrnl     journal.Journal
+	evtTypes [4]journal.EventType
```

Comment:
For performance reasons, an array of `journal.EventType`s is what I settled on as the lowest overhead, nicely encapsulated technique to initialize and store `EventTypes` during construction time, and index into them during runtime.